sunpy/net/dataretriever/sources/goes.py

```diff
@@ -168,9 +177,163 @@ def _can_handle_query(cls, *query):
         for x in query:
-            if x.__class__.__name__ == 'Instrument' and x.value.lower() in ('xrs', 'goes'):
+            if x.__class__.__name__ == "Instrument" and x.value.lower() in (
+                "xrs",
+                "goes",
+            ):
                 return all(chklist)
         return False
+
+
+class SUVIClient(GenericClient):
+
+    def _get_goes_sat_num(self, date):
+        """
+        Determines the best satellite number for a given date.
+
+        Parameters
+        ----------
+
+        date : `astropy.time.Time`
+            The date to determine which satellite is active.
+        """
+
+        # GOES-17 is operational but currently does not provide Level 2 data
+        # GOES-16 start date is based on the availability of regular level 1b data
+
+        suvi_operational = {
+            16: TimeRange("2018-06-01", parse_time("now")),
+        }
+
+        results = []
+        for sat_num in suvi_operational:
+            if date in suvi_operational[sat_num]:
+                # if true then the satellite with sat_num is available
+                results.append(sat_num)
+
+        if results:
+            # Return the newest satellite
+            return max(results)
+        else:
+            # if no satellites were found then raise an exception
+            raise ValueError(f"No operational SUVI instrument on {date.strftime(TIME_FORMAT)}")
+
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        Returns a URL to the SUVI data for the specified time range.
+
+        Parameters
+        ----------
+        timerange: sunpy.time.TimeRange
+            time range for which data is to be downloaded.
+        level : str
+            The level of the data.
+        wavelength :
+            Wavelength band.
+        satellitenumber : int
+            GOES satellite number. Must be >= 16.
+
+        """
+        base_url = "https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes{goes_number}/"
+        supported_waves = [94, 131, 171, 195, 284, 304]
+        supported_levels = ("2", "1b")
+
+        # these are optional requirements so if not provided assume defaults
+        # if wavelength is not provided assuming all of them
+        if "wavelength" in kwargs.keys():
+            wavelength_input = kwargs.get("wavelength")
+            if isinstance(wavelength_input, u.Quantity):  # not a range
+                if int(wavelength_input.to_value('Angstrom')) not in supported_waves:
+                    raise ValueError(f"Wavelength {kwargs.get('wavelength')} not supported.")
+                else:
+                    wavelength = [kwargs.get("wavelength")]
+            else:  #  _Range was provided
+                compress_index = [wavelength_input.wavemin <= this_wave <= wavelength_input.wavemax for this_wave in (supported_waves * u.Angstrom)]
+                if not any(compress_index):
+                    raise ValueError(
```

Comment:
I am not sure throwing a `ValueError` here is the correct thing to do. Doing this will kill outright any `Fido` query which misses the windows. Take for example the following query (where AIA has both and SUVI only has one passband):

```python
Fido.search(a.Instrument('SUVI') | a.Instrument('AIA'), a.Wavelength(30.4*u.nm) | a.Wavelength(211*u.AA))
```

This error here would cause it to fail, where as if it returned no results this query would work.

I think this is yet another case of the need to give users feedback on query validity, I just don't think this is the way to do it.

Reply:
I was just following what the `XRSClient` does. I'm not saying it's the right thing to do but since this PR is not meant to be controversial, I was just following the example of code that was already being used.